### PR TITLE
Add PutCommonInfo.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/123/
Test FAILed.

File: csclient/csclient_test.go [hunk 1]
```diff
@@ -1113,34 +1121,63 @@ func (s *suite) TestPutExtraInfo(c *gc.C) {
...
+		c.Assert(valCommonInfo.CommonInfo, jc.DeepEquals, info)
+	} else {
+		_, err = s.client.Meta(url, &valExtraInfo)
+		c.Assert(err, gc.IsNil)
+		c.Assert(valExtraInfo.ExtraInfo, jc.DeepEquals, info)
+	}
 }
 
-func (s *suite) TestPutExtraInfoWithError(c *gc.C) {
+func (s *suite) TestPutInfoWithError(c *gc.C) {
```
Comment (frankban) [inline csclient/csclient_test.go]: Let's keep these two tests separated.

**Comment by frankban** [comment]
:+1: with a trivial.
Also take a look at the weirdness on CI about dependencies format.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/124/
Test FAILed.

**Comment by mhilton** [comment]
:+1: when you get the tests working.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/125/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/126/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:  thanks @frankban  && @mhilton

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**